src/ape/utils.py

```diff
@@ -513,45 +557,106 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
+
+    chain_manager: ClassVar["ChainManager"] = cast("ChainManager", injected_before_use())
+
+    compiler_manager: ClassVar["CompilerManager"] = cast("CompilerManager", injected_before_use())
     """
-    A `data class <https://docs.python.org/3/library/dataclasses.html>`__ that
-    is also abstract (meaning it has methods that **must** be implemented in a
-    sub-class or else errors will occur). This class cannot be instantiated
-    on its own.
+    The group of compiler plugins for compiling source files. See
+    :class:`~ape.managers.compilers.CompilerManager` for more information.
+    Call method :meth:`~ape.managers.project.ProjectManager.load_contracts` in this class
+    to more easily compile sources.
     """
 
+    config_manager: ClassVar["ConfigManager"] = cast("ConfigManager", injected_before_use())
+    """
+    A reference to :class:`~ape.managers.config.ConfigManager`, which
+    manages project and plugin configurations.
+    """
 
-abstractdataclass = partial(dataclass, kwargs=True, meta=AbstractDataClassMeta)
-"""
-A `data class <https://docs.python.org/3/library/dataclasses.html>`__ that is
-also abstract (meaning it has methods that **must** be implemented or else
-errors will occur. This class cannot be instantiated on its own.
-"""
+    conversion_manager: ClassVar["ConversionManager"] = cast(
+        "ConversionManager", injected_before_use()
+    )
+    """
+    A reference to the conversion utilities in
+    :class:`~ape.managers.converters.ConversionManager`.
+    """
 
+    dependency_manager: ClassVar["_DependencyManager"] = cast(
+        "_DependencyManager", injected_before_use()
+    )
 
-class injected_before_use(property):
+    network_manager: ClassVar["NetworkManager"] = cast("NetworkManager", injected_before_use())
     """
-    Injected properties are injected class variables that must be set before use
-    NOTE: do not appear in a Pydantic model's set of properties
+    The manager of networks, :class:`~ape.managers.networks.NetworkManager`.
+    To get the active provide, use
+    :py:attr:`ape.managers.networks.NetworkManager.active_provider`.
     """
 
-    def __get__(self, *args):
-        raise ValueError("Value not set. Please inject this property before calling.")
+    plugin_manager: ClassVar["PluginManager"] = cast("PluginManager", injected_before_use())
+
+    project_manager: ClassVar["ProjectManager"] = cast("ProjectManager", injected_before_use())
+
+    query_manager: ClassVar["QueryManager"] = cast("QueryManager", injected_before_use())
+
+
+class AbstractBase(ManagerAccessBase, ABC):
+    """
+    Abstract class that has manager access.
+    """
+
+
+class AbstractBaseModel(AbstractBase, BaseModel):
+    """
+    Abstract class with manager access on a pydantic base model
+    """
+
+    class Config:
+        """
+        NOTE: Due to https://github.com/samuelcolvin/pydantic/issues/1241 we have
```

Comment:
This `NOTE` seems out of place? It is referring some cached property yet is a doc str on the entire class?

Reply:
Class doc strs should only be telling the user how to use the class and non-public classes should not have them.

Is this a public class?